python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vxlan-interface.j2

```diff
@@ -75,15 +75,27 @@
 
 | VLAN | VNI | Flood List | Multicast Group |
 | ---- | --- | ---------- | --------------- |
-{%         for vlan in vxlan_config.vxlan.vlans | arista.avd.natural_sort('id') %}
-{%             set vlan_vni = vlan.vni | arista.avd.default('-') %}
-{%             set multicast_group = vlan.multicast_group | arista.avd.default('-') %}
-{%             if vlan.flood_vteps is arista.avd.defined %}
-{%                 set flood_list = vlan.flood_vteps | join('<br/>')  %}
-{%             else %}
-{%                 set flood_list = '-' %}
-{%             endif %}
-| {{ vlan.id }} | {{ vlan_vni }} | {{ flood_list }} | {{ multicast_group }} |
+{%         set all_vlans = vxlan_config.vxlan.vlans | map(attribute='id') | map('arista.avd.range_expand') | sum(start=[]) %}
+{%         for all_vlans_item in all_vlans | arista.avd.natural_sort %}
+{%             for vxlan_vlans in vxlan_config.vxlan.vlans %}
+{%                 set vlans = vxlan_vlans.id | arista.avd.range_expand %}
+{%                 if all_vlans_item in vlans %}
+{%                     if vxlan_vlans.vni is arista.avd.defined %}
+{%                         set vlan_idx = vlans.index(all_vlans_item) %}
+{%                         set vlan_vnis = vxlan_vlans.vni | arista.avd.range_expand %}
+{%                         set vlan_vni = vlan_vnis[vlan_idx] %}
+{%                     else %}
+{%                         set vlan_vni = '-' %}
+{%                     endif %}
+{%                     set multicast_group = vxlan_vlans.multicast_group | arista.avd.default('-') %}
+{%                     if vxlan_vlans.flood_vteps is arista.avd.defined %}
+{%                         set flood_list = vxlan_vlans.flood_vteps | join('<br/>')  %}
+{%                     else %}
+{%                         set flood_list = '-' %}
+{%                     endif %}
+| {{ all_vlans_item }} | {{ vlan_vni }} | {{ flood_list }} | {{ multicast_group }} |
+{%                 endif %}
+{%             endfor %}
 {%         endfor %}
```

Comment:
it could be more optimized like this (discussed with @Vibhu-gslab)
```suggestion
{%         for vxlan_vlans in vxlan_config.vxlan.vlans %}
{%             set vlans = vxlan_vlans.id | arista.avd.range_expand %}
{%                 if vxlan_vlans.vni is arista.avd.defined %}
{%                     set vlan_vni = vxlan_vlans.vni | arista.avd.range_expand %}
{%                 endif %}
{%                 set multicast_group = vxlan_vlans.multicast_group | arista.avd.default('-') %}
{%                 if vxlan_vlans.flood_vteps is arista.avd.defined %}
{%                     set flood_list = vxlan_vlans.flood_vteps | join('<br/>')  %}
{%                 endif %}
{%             for idx in range(vlans | length) %}
| {{ vlans[idx] }} | {{ vlan_vni[idx] | arista.avd.default("-") }} | {{ flood_list | arista.avd.default("-") }} | {{ multicast_group }} |
{%             endfor %}
{%         endfor %}
```

Reply:
This is not sorting the vlan ids in documentation which is hard to read when there are large no of vlans. Also if you add one more item {id:113, multicast_group:239.1.1.1} for host one, it creates a new row for vlan 113.